tests/test_pagination.py

```diff
@@ -69,6 +79,7 @@ def test_run(self):
         self.assertEqual(set(), self.TESTABLE_STREAMS,
                          msg="Testable streams exist for this category.")
         print("\tThere are no testable streams.")
+        self.cleanup_excess()
```

Comment:
Can you put this in a teardown method? Probably in a https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDownClass method?

Reply:
I could use some help with this. I did put it in a teardown originally but it was running multiple times and I couldn't figure out why.

Reply:
I assume this is because of the issues mentioned earlier regarding the base class not extending testcase. It probably runs every method starting with the "test" prefix (if you just use tearDown). So tearDownClass should only run once, theoretically.

Reply:
Got this working I was using tearDown() before not the class method tearDownClass. I think that was my issue